admin/preprint_providers/views.py

```diff
@@ -367,3 +367,38 @@ def get_context_data(self, *args, **kwargs):
         kwargs['show_taxonomies'] = SHOW_TAXONOMIES_IN_PREPRINT_PROVIDER_CREATE
         kwargs['tinymce_apikey'] = settings.TINYMCE_APIKEY
         return super(CreatePreprintProvider, self).get_context_data(*args, **kwargs)
+
+
+class SharePreprintProviderWhitelistAdd(PermissionRequiredMixin, View):
+    permission_required = 'osf.change_preprintprovider'
```

Comment:
You'll need to add a custom migration to add this permission to the `osf_admin` Group. See this for an example: https://github.com/CenterForOpenScience/osf.io/blob/f575caf19c2b3b08d5f0b5636d03bae992a13f18/osf/migrations/0078_add_banner_permissions.py

Reply:
`change_preprintprovider` is an existing Permission that I reuse. I checked through osfshell locally and `osf_admin` group has this permission. It seems that a previous migration (https://github.com/CenterForOpenScience/osf.io/blob/f575caf19c2b3b08d5f0b5636d03bae992a13f18/osf/migrations/0030_preprint_provider_institution_permissions.py#L44) already adds this permission to the group.

Reply:
Ah, sure enough. Sorry, I thought this was a new permission. No action necessary then